bindings/node.js/kzg.cxx

```diff
@@ -126,19 +112,11 @@ Napi::Value FreeTrustedSetup(const Napi::CallbackInfo& info) {
 
 // blobToKzgCommitment: (blob: Blob, setupHandle: SetupHandle) => KZGCommitment;
 Napi::Value BlobToKzgCommitment(const Napi::CallbackInfo& info) {
-  auto env = info.Env();
-
-  size_t argument_count = info.Length();
-  size_t expected_argument_count = 2;
-  if (argument_count != expected_argument_count) {
-    return throw_invalid_arguments_count(expected_argument_count, argument_count, env);
-  }
-
-  Blob *blob = (Blob *)extract_byte_array_from_param(info, 0, "blob");
-  if (env.IsExceptionPending()) {
-    return env.Null();
+  Napi::Env env = info.Env();
```

Comment:
Hmm.. How come we don't need the argument count checking anymore? Was it useless before?

Reply:
It was definitely useful before to make sure the js functions passed the `KzgSettings *` back to the native side.  If for some reason one of the `requireSetupHandle` calls don't return the settings, [like on this line](https://github.com/matthewkeil/c-kzg-4844/blob/94198b5c18f8b82c2e56d79d7741ff583399f9e2/bindings/node.js/kzg.ts#L183), there will be a segfault [on this line](https://github.com/matthewkeil/c-kzg-4844/blob/94198b5c18f8b82c2e56d79d7741ff583399f9e2/bindings/node.js/kzg.cxx#L142) when trying to unwrap the `External` value.

There is no way to "validate" and external value other than the `IsExternal` method or by checking the number of arguments passed, although the latter technically doesn't protect against some other value getting passed in that argument position.  Which will throw a hard C error and not a JS exception.  Those are non-recoverable generally.

This is one of the main reasons for the [context-aware-addon](https://github.com/matthewkeil/c-kzg-4844/tree/context-aware-addon) branch that I mentioned yesterday and will work to get finalized PR for today.  It will accomplish a couple of things but storing that pointer in C is a biggie.  It wouldn't need to be [passed back](https://github.com/matthewkeil/c-kzg-4844/blob/94198b5c18f8b82c2e56d79d7741ff583399f9e2/bindings/node.js/kzg.cxx#L108) to JS, stored [in JS](https://github.com/matthewkeil/c-kzg-4844/blob/94198b5c18f8b82c2e56d79d7741ff583399f9e2/bindings/node.js/kzg.ts#L75) mutably and then passed back to the native code for [calling the functions](https://github.com/matthewkeil/c-kzg-4844/blob/94198b5c18f8b82c2e56d79d7741ff583399f9e2/bindings/node.js/kzg.ts#L212).

It would be possible to just store it in a static/global variable in C but that will crash a worker thread if someone implements the library using workers.  The safe "node" way to store that pointer is by attaching it to the node process and there are a few ways that Node provides to do that.

For this PR though, would you like me to add the argument length check back in and then also add the `IsExternal` call I mentioned in case that PR does not get merged?  When I broke this first piece out I did so assuming they would all get merged but thinking about that a bit more, its not a great plan, just in-case.  Having the `IsExternal` with a unit test to make sure the check is in all functions might be a really good idea...  What are your thoughts?

As a note, if it does get merged I will be able to pull those added length checks back out because we will be fully vetting all arguments to meet the function signature and the standard practice in JS is to allow for duck typing (if it walks like a duck... let it fly). Ie extra stuff that is passed to functions (arguments or even props/fields of objects) that are unused are left alone and do not raise an error so function chaining, or props getting passed through multiple functions work as expected in all of them.  While not necessarily applicable to this library it will be an option that I would like to put forward for the team to select.

Reply:
UPDATE: I put all length checks back as they were and added `IsExternal` checks before all `As<Napi:External>` calls